lib/internal/checkUrl.js

```diff
@@ -66,6 +85,15 @@ function checkUrl(link, baseUrl, options, callback)
 		}
 		
 		linkObj.clean(link);
+
+		if ( cache !== undefined && cache !== null )
```

Comment:
In the case of `null` and `undefined`, it doesn't run the whole non-strict operation (such as changing to string and number). It checks if `undefined` first, then `null` and stops because it is either one of those or not. Whereas `if (!cache)` would run many more checks.